### ISSUE TITLE
Nullable reference types in LdapEntry GetOrDefault-methods

### ISSUE THREAD

**Issue by PerssonP** [dsbenghe/Novell.Directory.Ldap.NETStandard#262]
Testing out 4.0.0-beta6 and having some small issues when using it with my own libraries.
I like the new "GetOrDefault"-methods (as discussed in #159) but I'm running into some trouble because this library does not currently use nullable contexts, which were introduced in C# 8 and are default enabled in projects starting with C# 10.

Consider a project with nullable context enabled that uses this library.
When using any of the new "GetOrDefault"-methods (and not supplying a not-null fallback) the return value could potentially be null. But because the library does not utilize nullable reference types the IDE will give misleading information to the user.
![image](https://github.com/user-attachments/assets/7e758370-61b3-499b-ae8c-da9fe3bbf081)

Have you considering implementing nullable reference types in this library?
Perhaps following something like https://learn.microsoft.com/en-us/dotnet/csharp/nullable-migration-strategies
Maybe just start with enabling nullable context in LdapEntry.cs for the benefit of the new GetOrDefault-methods? I could get a PR done for this if it's interesting.

**Comment by dsbenghe** [comment]
Yes, introducing nullable reference types in the library will ultimately happen. Happy to get PRs about it.
